course05/homework02/freekpaans/linked_list.cc

```diff
@@ -0,0 +1,74 @@
+#include <string>
+#include <numeric>
+
+#include "linked_list.h"
+
+static void push_at(LinkedList::LinkPtr& link, const std::string& str)
+{
+	LinkedList::LinkPtr newLink{new LinkedList::Link { str, {} }};
+
+	newLink->mNext = std::move(link);
+	link = std::move(newLink);
+}
+
+LinkedList::LinkedList(const LinkedList& from)
+{
+	std::accumulate(
```

Comment:
Nice that you use an algorithm, but I think this is confusing here; `std::accumulate` is defined in `numeric` to compute a sum!

You might want to look at `std::front_inserter`: https://en.cppreference.com/w/cpp/iterator/front_inserter

Reply:
I guess you have a bit of a narrower view of what a sum can be than I have :)

All jokes aside, this is actually just a left fold right? It's even documentated this way. So it's a bit weird it is in `numeric` IMO. Building up a collection with a left fold is quite a common pattern in more functional languages. Also, seems I'm not the only one using it for not-sums, look at this one https://github.com/CppCon/CppCon2016/blob/master/Presentations/accumulate%20-%20Exploring%20an%20Algorithmic%20Empire/accumulate%20-%20Exploring%20an%20Algorithmic%20Empire%20-%20Ben%20Deane%20-%20CppCon%202016.pdf slide 3.4/3.5 :)

`front_inserter` wouldn't work right, because that would result in reversed copy.